hanabi.py

```diff
@@ -119,44 +132,117 @@ def evaluate(self, turn: Turn):
         self.turns.append(turn)
         self.turn_index += 1
 
+    def _clue_turn(self, turn: Clue):
+        target = turn.target
+        logging.info(f"Player {self.current_player} clued player {target} {turn.color or turn.number}")
+
+        if target == self.current_player:
+            raise InvalidMove("cannot clue self")
+        if self.clues == 0:
+            raise InvalidMove("no clues available")
+
+        # Update Information
+        logging.info(f"Player {target} Old Information: {[str(x) for x in self.information[target]]}")
+        for idx, card in enumerate(self._hands[target]):
+            if turn.color:
+                if card.color == turn.color:
+                    self.information[target][idx].color.clear()
+                    self.information[target][idx].color.add(turn.color)
+                    self.information[target][idx].clued = True
+                else:
+                    self.information[target][idx].color.discard(turn.color)
+
+            if turn.number:
+                if card.number == turn.number:
+                    self.information[target][idx].number.clear()
+                    self.information[target][idx].number.add(turn.number)
+                    self.information[target][idx].clued = True
+                else:
+                    self.information[target][idx].number.discard(turn.number)
```

Comment:
Unless I'm missing something, the method for removing from a set should be `.remove` not `discard`?

Reply:
.discard() succeeds if the element isn't present as opposed to .remove() which throws an exception.